### PR TITLE
Test PennyLane tensor unwrapping

### PR DESCRIPTION
**Pull request by antalszava**
**Context**

This PR adds tests from https://github.com/PennyLaneAI/pennylane/pull/903 using the `qiskit.aer` device. 

Also, see the discussion in https://github.com/PennyLaneAI/pennylane/pull/893 for further details.

These tests cover examples that would be failing with PennyLane version `<0.12.0`:
```python
import pennylane as qml
from pennylane.numpy import tensor

dev = qml.device('qiskit.aer', wires=4)
@qml.qnode(dev)
def circuit(phi=None):
    for y in phi:
        for idx, x in enumerate(y):
            qml.RX(x, wires=idx)
    return qml.expval(qml.PauliZ(0))
phi = tensor([[0.04439891, 0.14490549, 3.29725643, 2.51240058]])
circuit(phi=phi)
```
```
~/anaconda3/envs/pl_qiskit/lib/python3.7/site-packages/qiskit/providers/aer/backends/backend_utils.py in cpp_execute(controller, qobj)
     49     qobj_dict['config']['library_dir'] = LIBRARY_DIR
     50 
---> 51     return controller(qobj_dict)
     52 
     53 

RuntimeError: Invalid number of dimensions!
```
**Passing checks - master of PennyLane**
See the passing checks for commit https://github.com/PennyLaneAI/pennylane-qiskit/commit/72b8fe172fcddfead7e1ce4764cdfdac7b8da444 with the master version of PennyLane `git+https://github.com/PennyLaneAI/pennylane.git`: https://github.com/PennyLaneAI/pennylane-qiskit/actions/runs/385779826.

**Failing checks -- PennyLane v0.12.0**
Commit https://github.com/PennyLaneAI/pennylane-qiskit/commit/cbbb4be3f37b5b6b8f962601cb55f862480e4027 and check https://github.com/PennyLaneAI/pennylane-qiskit/runs/1460625139.

### CODE REVIEW

**Comment by antalszava** [comment]
The checks passed for commit https://github.com/PennyLaneAI/pennylane-qiskit/commit/72b8fe172fcddfead7e1ce4764cdfdac7b8da444 with the master version of PennyLane `git+https://github.com/PennyLaneAI/pennylane.git`: https://github.com/PennyLaneAI/pennylane-qiskit/actions/runs/385779826.

File: requirements.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 qiskit>=0.23.1
-pennylane>=0.12.0
+git+https://github.com/PennyLaneAI/pennylane.git
```
Comment (antalszava) [inline requirements.txt:2]: This is adjusted in #117 to `pennylane>=0.13.0`.

Comment (antalszava) [inline requirements.txt:2]: This is adjusted in #117 to `pennylane>=0.13.0`.

**Comment by antalszava** [comment]
Checks are passing as we pull the master version of PennyLane. They fail if don't bump the PL version. Therefore the test cases indeed cover the missing edge cases fixed in the latest of PL.

**Failing checks -- PennyLane v0.12.0**
Commit https://github.com/PennyLaneAI/pennylane-qiskit/commit/cbbb4be3f37b5b6b8f962601cb55f862480e4027 and check https://github.com/PennyLaneAI/pennylane-qiskit/runs/1460625139.

**Passing checks - master of PennyLane**
See the passing checks for commit https://github.com/PennyLaneAI/pennylane-qiskit/commit/72b8fe172fcddfead7e1ce4764cdfdac7b8da444 with the master version of PennyLane `git+https://github.com/PennyLaneAI/pennylane.git`: https://github.com/PennyLaneAI/pennylane-qiskit/actions/runs/385779826.

File: tests/test_integration.py [hunk 2]
```diff
@@ -311,6 +312,138 @@ def basisstate():
...
+class TestPLTemplates:
+    """Integration tests for checking certain PennyLane templates."""
+
+    def test_random_layers_tensor_unwrapped(self, monkeypatch):
+        """Test that if random_layer() receives a one element PennyLane tensor,
+        then it is unwrapped successfully.
+
+        The test involves using RandomLayers, which then calls random_layer
+        internally. Eventually each gate used by random_layer receives a single
+        scalar.
```
Comment (josh146) [inline tests/test_integration.py:324]: :+1:

File: tests/test_integration.py [hunk 3]
```diff
@@ -311,6 +312,138 @@ def basisstate():
...
+            m.setattr(qml.templates.layers.random, "RX", mock_func)
+            m.setattr(qml.templates.layers.random, "RY", mock_func)
+            m.setattr(qml.templates.layers.random, "RZ", mock_func)
+
+            @qml.qnode(dev)
+            def circuit(phi=None):
+                qml.templates.layers.RandomLayers(phi, wires=list(range(4)))
+                return qml.expval(qml.PauliZ(0))
+
+            phi = qml.numpy.tensor([[0.04439891, 0.14490549, 3.29725643, 2.51240058]])
```
Comment (josh146) [inline tests/test_integration.py:345]:
```suggestion
            # RandomLayers loops over the random_layer function, with each call to random_layer
            # being passed a `np.tensor` scalar.
            phi = qml.numpy.tensor([[0.04439891, 0.14490549, 3.29725643, 2.51240058]])
```

File: tests/test_integration.py [hunk 4]
```diff
@@ -311,6 +312,138 @@ def basisstate():
...
+
+        with qml._queuing.OperationRecorder() as rec:
+            circuit(phi=phi)
+
+        # Test the rotation applied
+        assert rec.queue[0].name == "Rot"
+        assert len(rec.queue[0].parameters) == 3
+
+        # Test that the gate parameters are not PennyLane tensors, but a
+        # floats
```
Comment (josh146) [inline tests/test_integration.py:437]:
```suggestion
        # Test that the gate parameters are not PennyLane tensors,
        # but are instead floats
```

**Review by josh146 — APPROVED** [review]
Very thorough testing Antal!
